docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,231 @@
+---
+title: What's new in C# 9.0 - C# Guide
+description: Get an overview of the new features available in C# 9.0.
+ms.date: 09/04/2020
+---
+# What's new in C# 9.0
+
+C# 9.0 adds the following features and enhancements to the C# language:
+
+- Records
+- Init only setters
+- Top-level statements
+- Pattern matching enhancements
+- Native sized integers
+- Function pointers
+- Suppress emitting localsinit flag
+- Target-typed new expressions
+- static anonymous functions
+- Target-typed conditional expressions
+- Covariant return types
+- Lambda discard parameters
+- Attributes on local functions
+- Module initializers
+- New features for partial methods
+
+C# 9.0 is supported on **.NET 5**. For more information, see [C# language versioning](../language-reference/configure-language-version.md).
+
+## Record types
+
+C# 9.0 introduces ***record types***, which are a reference type that provides synthesized methods to provide value semantics for equality. Records are immutable by default.
+
+Record types make it easy to create immutable reference types in .NET. Historically, .NET types are largely classified as reference types (including classes and anonymous types) and value types (including structs and tuples). While immutable value types are recommended, mutable value types don’t often introduce errors. Value types are passed by value to methods so changes are made to a copy of the original data.
```

Comment:
nit:

```suggestion
Record types make it easy to create immutable reference types in .NET. Historically, .NET types are largely classified as reference types (including classes and anonymous types) and value types (including structs and tuples). While immutable value types are recommended, mutable value types don't often introduce errors. Value types are passed by value to methods so changes are made to a copy of the original data.
```

Reply:
I think that *technically*, both reference types and value types are passed by value. However, the *value* of a passed reference type is a pointer in the heap.

The last sentence probably implies that reference types are passed by reference by default (which seem incorrect to me).

Certainly, the sentence doesn't say that reference types are passed by reference. So the sentence is completely correct. However, I think readers may interpret it incorrectly.